docs/sources/developer/architecture/components/api.md

```diff
@@ -0,0 +1,78 @@
+# API
+
+The Faro API is the primary interface for end-users to interact with Faro. It is the central place where signals are
+sent to but it also contains helpers methods for other components like metas.
+
+## Signals
+
+### Events
+
+Events are signals related to tracking various actions like mouse clicks as well as other events like transitioning to
+another page.
+
+Methods:
+
+- `pushEvent` - sends an event
+
+### Exceptions
+
+Exceptions are signals that are represented by errors which occurred in an app.
+
+When an exception is sent, the API will pass it through a stacktrace parser to extract the relevant information from it.
+By default, the core package does not contain any stacktrace parsers and they are either provided by wrapper packages
+like `web-sdk` or by the user.
+
+Methods:
+
+- `changeStacktraceParser` - replaces the existing stacktrace parser with a new one
+- `getStacktraceParser` - returns the current stacktrace parser
+- `pushError` - sends an exception
+
+### Logs
+
+Logs are signals which capture events that happen within the app itself. They do not have a business meaning but a
+technical meaning.
+
+Methods:
+
+- `pushLog` - sends a log
+
+### Measurements
+
+Measurements are signals which can be seen as metrics.
+
+Methods:
+
+- `pushMeasurement` - sends a measurement
+
+### Traces
+
+Traces are a special type of signal which are used to track the behaviour of an app. Because Faro does not have an
+internal tracing mechanism, we rely on OpenTelemetry's tracing mechanism to provide this functionality. However,
+OpenTelemetry is quite a heavy system and we do not want to force users to use it. Therefore, the traces API is
+pretty simple because all the logic is externalized to other packages like `web-tracing`.
```

Comment:
nit: what are we trying to convey here? Maybe we can link to a document that explains "tracing api is via OTel to provide maximum flexibility to users. see here for otel tracing information, or here (link to otel tracing providers such as web-tracing(?)).

Reply:
This will come in a separate document. Currently we present only the core components (aka `core` package).